scripts/jenkins/Jenkinsfile

```diff
@@ -0,0 +1,567 @@
+/* https://build.libelektra.org/jenkins/job/elektra-jenkinsfile/
+ * This file describes how the elektra-jenkinsfile buildjob should be
+ * executed.
+ *
+ * 1. imports and global variables are set
+ * 2. define the main stages of the pipeline
+ * 3. describe sub stages. This is where you will want to add new builds
+ * 4. helper section to help write build scripts
+ */
+
+// FIXME sloc publish errors
+// XXX add missing jobs
+// TODO have a per plugin/binding deps in Dockerfile for easier maintanence
+// TODO add warnings plugins to scan for compiler warnings
+
+// Imports
+import java.text.SimpleDateFormat
+
+// Buildjob properties
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '60', artifactNumToKeepStr: '60'))
+])
+
+// Globals
+docker_node_label = 'docker'
+docker_image_prefix = 'build-elektra-'
+
+DOCKER_IMAGES = [
+    sid: [
+        id: docker_image_prefix + 'debian-sid',
+        context: "./scripts/docker/debian/sid",
+        file: "./scripts/docker/debian/sid/Dockerfile"
+    ],
+    stretch: [
+        id: docker_image_prefix + 'debian-stretch',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile",
+    ],
+    stretch_minimal: [
+        id: docker_image_prefix + 'debian-stretch-minimal',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile.minimal",
+    ],
+    stretch_doc: [
+        id: docker_image_prefix + 'debian-stretch-doc',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile.doc",
+    ],
+    jessie: [
+        id: docker_image_prefix + 'debian-jessie',
+        context: "./scripts/docker/debian/jessie",
+        file: "./scripts/docker/debian/jessie/Dockerfile",
+    ],
+    xenial: [
+        id: docker_image_prefix + 'ubuntu-xenial',
+        context: "./scripts/docker/ubuntu/xenial",
+        file: "./scripts/docker/ubuntu/xenial/Dockerfile",
+    ]
+]
+
+// Define reusable cmake Flag globals
```

Comment:
why do we need globals here? what are they for? when would I need to declare them?

Reply:
I will not continue working  on this because I will have  to redo it in the next PR.

If i want to add something that is similar to the current matrix builds I will need another way to describe possible values of an configuration anyway.